test/unit/event.js

```diff
@@ -1852,6 +1852,14 @@ QUnit.test( "delegated event with intermediate DOM manipulation (#13208)", funct
 	jQuery( "#anchor2" ).trigger( "click" );
 } );
 
+QUnit.test( "ignore comment nodes in event delegation (gh-2055)", function( assert ) {
+	assert.expect( 1 );
+
+	jQuery( "#foo" ).on( "DOMNodeInserted", "[id]", function() {
+		assert.ok( true, "No error thrown on comment node" );
+	} ).find( "#sap" ).append( "<!-- comment -->" );
```

Comment:
The deprecation of `DOMNodeInserted` event could be addressed by a plan B that uses [fireNative](https://github.com/jquery/jquery/blob/0fb84fa8ccefcd07febf282fd7b80262ad70add7/test/data/testinit.js#L114-L123) to synthesize an event on a comment node.